src/main/java/com/coveo/saml/ResponseSchemaValidator.java

```diff
@@ -0,0 +1,45 @@
+package com.coveo.saml;
+
+import org.apache.commons.lang3.StringUtils;
+import org.opensaml.saml.common.SAMLVersion;
+import org.opensaml.saml.saml2.core.Response;
+
+import java.util.Objects;
+
+import javax.xml.bind.ValidationException;
+
+public class ResponseSchemaValidator {
```

Comment:
I'm guessing this is also borrowed from OpenSAML?

Reply:
Correct. :)